### DISCUSSION TITLE
Feature request

### DISCUSSION THREAD

**Discussion by mixlplix** [oqtane/oqtane.framework#1870] [Ideas]
It looks like Oqtane leverages Microsoft.AspNetIdentity for the AspNetUsers table, I'm wondering if the standard account lockout features are configurable?

Also wondering if sign-in success/failure can be logged.  

We have a client who is requesting both of these features for security purposes.

**Comment by hishamco** [comment]
Oqtane uses Identity, you can configure identity options through `IdentityOptions` in startup, for lockout you can use normal manager APIs that available in Identity APIs, I'm not quite sure if users APIs in Oqtane provide all the features, but you could have a look

**Comment by hishamco** [comment]
Oqtane uses Identity, you can configure identity options through `IdentityOptions` in startup, for lockout you can use normal manager APIs that available in Identity APIs, I'm not quite sure if users APIs in Oqtane provide all the features, but you could have a look

**Comment by leigh-pointer** [comment]
@mixlplix signin is logged.  If you check the logs module in the Admin dashboard you will see the log in and failure are logged.

**Comment by mixlplix** [comment]
Excellent, thank you.  I do see that now.

**Comment by mixlplix** [comment]
So, is there a way to override those options using IServerStartup, or would we have to fork the repo to make changes directly in Startup.cs?

**Comment by leigh-pointer** [comment]
Check out https://github.com/oqtane/oqtane.framework/discussions/1008
________________________________

**Reply by leigh-pointer** [reply]
You can create a class in your module's Server project and add the service registration logic you would normally add in the Startup class. During the startup Oqtane will call the methods automatically.

```
    public class ServerStartup : IServerStartup
    {
        public void ConfigureServices(IServiceCollection services)
        {            
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
        }

        public void ConfigureMvc(IMvcBuilder mvcBuilder)
        {
        }
    }
```

**Comment by sbwalker** [comment]
@mixlplix currently account lockout is not currently enabled in Oqtane. The configuration is specified in Startup:

```
                // Lockout settings
                options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(30);
                options.Lockout.MaxFailedAccessAttempts = 10;
                options.Lockout.AllowedForNewUsers = true;
```
However in order to use lockout you need to set the lockoutOnFailure parameter to True when calling the SigninManager.CheckPasswordSignInAsync() method. Currently Oqtane sets this parameter to False when it calls the method. The only way to enable it currently would be to fork the repo and modify the logic in the applicable areas. 

We can certainly add this to the roadmap for a future version - but its more than simply changing False to True. It should be configurable for each Site. And when the lockout event occurs, the best practice is to send a notification to the user to let them know.

**Comment by mixlplix** [comment]
OK, thanks
